src/raftstore/store/snap.rs

```
@@ -2209,30 +1135,21 @@ impl SnapManager {
         let core = self.core.rl();
         let mut snapshot_data = RaftSnapshotData::new();
         try!(snapshot_data.merge_from_bytes(data));
-        if snapshot_data.get_version() == v2::SNAPSHOT_VERSION {
-            let f = try!(v2::Snap::new_for_receiving(&core.base,
-                                                     key,
-                                                     snapshot_data.take_meta(),
-                                                     core.snap_size.clone(),
-                                                     Box::new(self.clone())));
-            Ok(Box::new(f))
-        } else {
-            let f = try!(v1::Snap::new_for_writing(&core.base, core.snap_size.clone(), false, key));
-            Ok(Box::new(f))
-        }
+        let f = try!(Snap::new_for_receiving(&core.base,
+                                             key,
+                                             snapshot_data.take_meta(),
+                                             core.snap_size.clone(),
+                                             Box::new(self.clone())));
+        Ok(Box::new(f))
     }
 
     pub fn get_snapshot_for_applying(&self, key: &SnapKey) -> RaftStoreResult<Box<Snapshot>> {
         let core = self.core.rl();
-        if let Ok(s) = v2::Snap::new_for_applying(&core.base,
-                                                  key,
-                                                  core.snap_size.clone(),
-                                                  Box::new(self.clone())) {
-            if s.exists() {
-                return Ok(Box::new(s));
-            }
-        }
-        let s = try!(v1::Snap::new_for_reading(&core.base, core.snap_size.clone(), false, key));
+        let s = Snap::new_for_applying(&core.base,
+                                       key,
+                                       core.snap_size.clone(),
+                                       Box::new(self.clone()))?;
+        assert!(s.exists());
```

Can we just return an error when snapshot not exist?